src/vivarium_public_health/disease/model.py

```diff
@@ -70,20 +74,30 @@ def get_csmr(self):
         return self._csmr_data
 
     @staticmethod
-    def assign_initial_status_to_simulants(simulants_df, states, initial_state, randomness):
+    def assign_initial_status_to_simulants(simulants_df, states, initial_state, propensities):
         simulants = simulants_df[['age', 'sex']].copy()
         sequelae, weights = zip(*states.items())
         sequelae += (initial_state,)
         for w in weights:
             w.reset_index(inplace=True, drop=True)
         weights += ((1 - np.sum(weights, axis=0)),)
-        simulants.loc[:, 'condition_state'] = randomness.choice(simulants.index, sequelae,
-                                                                np.array(weights).T)
+
+        # manually calculate the assigned initial states in order to use propensities
```

Comment:
I think this function is now probably doing too much work.  Maybe we can make a `self.get_state_map` function that gets 
called on line 103 instead of the dict comprehension that handles a lot of this weight normalization up front.  Up to you.